core/src/main/java/org/javarosa/core/model/condition/Recalculate.java

```diff
@@ -59,27 +39,17 @@ public Object eval(FormInstance model, EvaluationContext ec) {
         }
     }
 
+    @Override
     public void apply(TreeReference ref, Object result, FormInstance model, FormDef f) {
         int dataType = f.getMainInstance().resolveReference(ref).getDataType();
         f.setAnswer(wrapData(result, dataType), ref);
     }
 
+    @Override
     public boolean canCascade() {
         return true;
     }
 
-    public boolean equals(Object o) {
-        if (o instanceof Recalculate) {
-            Recalculate r = (Recalculate)o;
-            if (this == r)
-                return true;
-
-            return super.equals(r);
-        } else {
-            return false;
-        }
-    }
-
```

Comment:
`super` implementation also does referential equality checks, so there is no need for this implementation.